src/Compilers/CSharp/Test/Syntax/Parsing/StatementParsingTests.cs

```diff
@@ -2740,6 +2774,185 @@ public void NullExceptionInLabeledStatement()
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, "else").WithLocation(1, 23),
+                // (1,23): error CS1525: Invalid expression term 'else'
+                // { if (a) { } else { } else { } }
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "else").WithArguments("else").WithLocation(1, 23),
+                // (1,23): error CS1002: ; expected
+                // { if (a) { } else { } else { } }
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "else").WithLocation(1, 23)
+                );
+            N(SyntaxKind.Block);
+            {
+                N(SyntaxKind.OpenBraceToken);
+                N(SyntaxKind.IfStatement);
+                {
+                    N(SyntaxKind.IfKeyword);
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "a");
+                    }
+                    N(SyntaxKind.CloseParenToken);
+                    N(SyntaxKind.Block);
+                    {
+                        N(SyntaxKind.OpenBraceToken);
+                        N(SyntaxKind.CloseBraceToken);
+                    }
+                    N(SyntaxKind.ElseClause);
+                    {
+                        N(SyntaxKind.ElseKeyword);
+                        N(SyntaxKind.Block);
+                        {
+                            N(SyntaxKind.OpenBraceToken);
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                }
+                N(SyntaxKind.IfStatement);
+                {
+                    M(SyntaxKind.IfKeyword);
+                    M(SyntaxKind.OpenParenToken);
+                    M(SyntaxKind.IdentifierName);
+                    {
+                        M(SyntaxKind.IdentifierToken);
+                    }
+                    M(SyntaxKind.CloseParenToken);
+                    M(SyntaxKind.ExpressionStatement);
+                    {
+                        M(SyntaxKind.IdentifierName);
+                        {
+                            M(SyntaxKind.IdentifierToken);
+                        }
+                        M(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.ElseClause);
+                    {
+                        N(SyntaxKind.ElseKeyword);
+                        N(SyntaxKind.Block);
+                        {
+                            N(SyntaxKind.OpenBraceToken);
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                }
+                N(SyntaxKind.CloseBraceToken);
+            }
+            EOF();
+        }
+
+        [WorkItem(27866, "https://github.com/dotnet/roslyn/issues/27866")]
+        [Fact]
+        public void ParseElseKeywordPlacedAsIfEmbeddedStatement()
+        {
+            UsingStatement("if (a) else {}",
```

Comment:
can we have an "else { } else { }" as well?

Reply:
There is `ParseSubsequentElseWithoutPrecedingIfStatement` already with its `UsingStatement("{ if (a) { } else { } else { } }"`. Do you mean `UsingStatement("{ if (a) else { } else { } }"`?

Reply:
Nope.  just ```else { } else { }``` :)

Reply:
Here you go: https://github.com/dotnet/roslyn/pull/28115/commits/b17e8b7396e6f1e36f3f7e69cbe99a19700923a4